numaprom/udsink/train_rollout.py

```diff
@@ -64,7 +64,8 @@ def _train_model(uuid, x, model_cfg, trainer_cfg):
     )
 
     train_reconerr = trainer.predict(model, dataloaders=DataLoader(dataset, batch_size=64))
-    return train_reconerr.numpy(), model
+    # return the trainer to avoid Weakreference error
+    return train_reconerr.numpy(), model, trainer
```

Comment:
was it getting garbage collected too early?

Reply:
yes something like that because trainer takes in the model, and does something to it. If we release the trainer variable, then the model gets weak-referenced.
Something similar here: https://github.com/Lightning-AI/lightning/issues/12233#issuecomment-1059484812